runtime/runtime.m

```diff
@@ -1104,52 +1104,15 @@ -(void) xamarinSetFlags: (enum XamarinGCHandleFlags) flags;
  * Exception handling
  */
 
-static MonoObject *
-fetch_exception_property (MonoObject *obj, const char *name, bool is_virtual)
-{
-	// COOP: reading managed memory and executing managed code: must be in UNSAFE mode
-	MONO_ASSERT_GC_UNSAFE;
-	
-	MonoMethod *get = NULL;
-	MonoMethod *get_virt = NULL;
-	MonoObject *exc = NULL;
-
-	get = mono_class_get_method_from_name (mono_get_exception_class (), name, 0);
-	if (get) {
-		if (is_virtual) {
-			get_virt = mono_object_get_virtual_method (obj, get);
-			if (get_virt)
-				get = get_virt;
-		}
-
-		return (MonoObject *) mono_runtime_invoke (get, obj, NULL, &exc);
-	} else {
-		PRINT ("Could not find the property System.Exception.%s", name);
-	}
-	
-	return NULL;
-}
-
-static char *
-fetch_exception_property_string (MonoObject *obj, const char *name, bool is_virtual)
-{
-	// COOP: reading managed memory and executing managed code: must be in UNSAFE mode
-	MONO_ASSERT_GC_UNSAFE;
-	
-	MonoString *str = (MonoString *) fetch_exception_property (obj, name, is_virtual);
-	return str ? mono_string_to_utf8 (str) : NULL;
-}
-
 static void
-print_exception (MonoObject *exc, bool is_inner, NSMutableString *msg)
+print_exception (GCHandle exception_gchandle, bool is_inner, NSMutableString *msg)
 {
 	// COOP: reading managed memory and executing managed code: must be in UNSAFE mode
 	MONO_ASSERT_GC_UNSAFE;
 	
-	MonoClass *type = mono_object_get_class (exc);
-	char *type_name = xamarin_strdup_printf ("%s.%s", mono_class_get_namespace (type), mono_class_get_name (type));
-	char *trace = fetch_exception_property_string (exc, "get_StackTrace", true);
-	char *message = fetch_exception_property_string (exc, "get_Message", true);
+	char *type_name = xamarin_get_object_type_fullname (exception_gchandle);
+	char *trace = xamarin_get_exception_message (exception_gchandle);
+	char *message = xamarin_get_exception_stacktrace (exception_gchandle);
```

Comment:
Not that it particularly matters but should the `mono_free` calls below be changed to `xamarin_free`?

Reply:
@filipnavara I've changed it as you suggested. Thanks!